docs/concepts/route-layout.md

```diff
@@ -0,0 +1,28 @@
+---
+description: |
+  Add a route layout to provide common meta tags, context for application sub routes, and common layout.
+---
+
+An route layout is defined in a `_layout.tsx` file in any sub directory (at any
+level) under the `routes/` folder. It must contain a default export that is a
+regular Preact component. Only one such layout is allowed per sub directory.
+
+The component to be wrapped is received via props, in addition to a few other
+things. This allows for the introduction of a global container functioning as a
+template which can be conditioned based on state and params. Note that any state
+set by middleware is available via `props.state`.
+
+```tsx
+// routes/sub/_layout.tsx
+
+import { LayoutProps } from "$fresh/server.ts";
+
+export default function Layout({ Component, state }: LayoutProps) {
+  //do something with state here
+  return (
+    <div class="layout">
+      <Component />
```

Comment:
I guess the aim of this here is to match the `_app.tsx` format of a `Component` being passed as a prop & that be the slot point to hand off to the child component(s).

However is there any argument or reason why something more like [Next.js' style ](https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts#layouts)of just having a `children` prop instead? The only thought I can think for the former being thinking of prop drilling but I don't think that really applies here does it?

Reply:
The `children` prop feels a little wrong to me in this case. I look at the `children` prop as a tool for use when creating a custom component (server/island). Its a property of the custom component, and signifies where the "inner" contents of the element go <CustomComponent>{...children}</CustomComponent>.  

In the case of a Page, template (_app), or layout (_layout) there isn't actually "children" of the component, but rather the rendering of a component based on data that comes from outside of the properties. For this reason, passing in the `Component` feels like the right API as the children are unknown, but the inner Component is known.

Reply:
+1 on going with `children`. That's the way [nextjs does it too](https://nextjs.org/docs/pages/building-your-application/routing/pages-and-layouts) and to me it seems worth it to be familiar to users coming from other frameworks.

I don't think there is a strong distinction from Preact's point of view between a layout, a page or a component. For Preact all of these are standard components.

Reply:
I'm not as familiar with NextJS, so can't speak much to that. However, i do think that going with `children` is semantically different than the current framework approach to an Application wrapper.  As a user, it would feel awkward to me to use injected `<Component />` in the app wrapper and then to use `children` in a route layout.

So then the question is, would we want to change App wrapper to use children as well? So we could in theory be inline with those migrating from NextJS... or do we keep Route Layouts inline with the existing App wrapper implementation.  My +1 is towards a similar API to what we already have in App wrapper.

Reply:
That's a good point. I hadn't thought about there being `Component` already in `_app.tsx` and it makes sense to stay consistent. For me that resolves the discussion. Let's go with `Component`

Reply:
@lino-levan I think remix use an `<Outlet />` component which is kinda similar to our `<Component />` for that.

Yeah with layouts having `_app.tsx` is a kinda the same thing and becomes unnecessary. We should consider nudging people to a top level `_layout.tsx` file instead. I'd still only soft deprecate `_app.tsx` though by printing a warning or something. That way the upgrade process is easier and we don't have to do a breaking change. This imo should be done in a separate PR though.

Reply:
A separate PR makes sense to me. Printing a message for a few updates of fresh stating that it's deprecated and will be removed "at some point in the future" is probably good enough.

Reply:
To close this discussion: We'll go with `<Component />` for now and we can easily add support for `props.children` on top of it later.